New Label: Orbstack

jnichols35 opened the pull request:
❯ sudo ./Installomator.sh orbstack DEBUG=0
Password:
2023-06-07 23:11:19 : REQ   : orbstack : ################## Start Installomator v. 10.5beta, date 2023-06-07
2023-06-07 23:11:19 : INFO  : orbstack : ################## Version: 10.5beta
2023-06-07 23:11:19 : INFO  : orbstack : ################## Date: 2023-06-07
2023-06-07 23:11:19 : INFO  : orbstack : ################## orbstack
2023-06-07 23:11:19 : DEBUG : orbstack : DEBUG mode 1 enabled.
2023-06-07 23:11:19 : INFO  : orbstack : SwiftDialog is not installed, clear cmd file var
2023-06-07 23:11:20 : INFO  : orbstack : setting variable from argument DEBUG=0
2023-06-07 23:11:20 : DEBUG : orbstack : name=OrbStack
2023-06-07 23:11:20 : DEBUG : orbstack : appName=
2023-06-07 23:11:20 : DEBUG : orbstack : type=dmg
2023-06-07 23:11:20 : DEBUG : orbstack : archiveName=
2023-06-07 23:11:20 : DEBUG : orbstack : downloadURL=https://cdn-updates.orbstack.dev/amd64/OrbStack_v0.11.3_1648_amd64.dmg
2023-06-07 23:11:20 : DEBUG : orbstack : curlOptions=
2023-06-07 23:11:20 : DEBUG : orbstack : appNewVersion=
2023-06-07 23:11:20 : DEBUG : orbstack : appCustomVersion function: Not defined
2023-06-07 23:11:20 : DEBUG : orbstack : versionKey=CFBundleShortVersionString
2023-06-07 23:11:20 : DEBUG : orbstack : packageID=
2023-06-07 23:11:20 : DEBUG : orbstack : pkgName=
2023-06-07 23:11:20 : DEBUG : orbstack : choiceChangesXML=
2023-06-07 23:11:20 : DEBUG : orbstack : expectedTeamID=HUAQ24HBR6
2023-06-07 23:11:20 : DEBUG : orbstack : blockingProcesses=
2023-06-07 23:11:20 : DEBUG : orbstack : installerTool=
2023-06-07 23:11:20 : DEBUG : orbstack : CLIInstaller=
2023-06-07 23:11:20 : DEBUG : orbstack : CLIArguments=
2023-06-07 23:11:20 : DEBUG : orbstack : updateTool=
2023-06-07 23:11:20 : DEBUG : orbstack : updateToolArguments=
2023-06-07 23:11:20 : DEBUG : orbstack : updateToolRunAsCurrentUser=
2023-06-07 23:11:20 : INFO  : orbstack : BLOCKING_PROCESS_ACTION=tell_user
2023-06-07 23:11:20 : INFO  : orbstack : NOTIFY=success
2023-06-07 23:11:20 : INFO  : orbstack : LOGGING=DEBUG
2023-06-07 23:11:20 : INFO  : orbstack : LOGO=/System/Applications/App Store.app/Contents/Resources/AppIcon.icns
2023-06-07 23:11:20 : INFO  : orbstack : Label type: dmg
2023-06-07 23:11:20 : INFO  : orbstack : archiveName: OrbStack.dmg
2023-06-07 23:11:20 : INFO  : orbstack : no blocking processes defined, using OrbStack as default
2023-06-07 23:11:20 : DEBUG : orbstack : Changing directory to /var/folders/zz/zyxvpxvq6csfxvn_n0000000000000/T/tmp.nZyoVqJb
2023-06-07 23:11:20 : INFO  : orbstack : App(s) found: /Applications/OrbStack.app
2023-06-07 23:11:20 : INFO  : orbstack : found app at /Applications/OrbStack.app, version 0.11.3, on versionKey CFBundleShortVersionString
2023-06-07 23:11:20 : INFO  : orbstack : appversion: 0.11.3
2023-06-07 23:11:20 : INFO  : orbstack : Latest version not specified.
2023-06-07 23:11:20 : REQ   : orbstack : Downloading https://cdn-updates.orbstack.dev/amd64/OrbStack_v0.11.3_1648_amd64.dmg to OrbStack.dmg
2023-06-07 23:11:20 : DEBUG : orbstack : No Dialog connection, just download
2023-06-07 23:11:25 : DEBUG : orbstack : File list: -rw-r--r--  1 root  wheel   180M Jun  7 23:11 OrbStack.dmg
2023-06-07 23:11:25 : DEBUG : orbstack : File type: OrbStack.dmg: lzfse encoded, lzvn compressed
2023-06-07 23:11:25 : DEBUG : orbstack : curl output was:
*   Trying [2606:4700:3032::ac43:90d2]:443...
* Connected to cdn-updates.orbstack.dev (2606:4700:3032::ac43:90d2) port 443 (#0)
* ALPN: offers h2,http/1.1
* (304) (OUT), TLS handshake, Client hello (1):
} [329 bytes data]
*  CAfile: /etc/ssl/cert.pem
*  CApath: none
* (304) (IN), TLS handshake, Server hello (2):
{ [122 bytes data]
* (304) (IN), TLS handshake, Unknown (8):
{ [25 bytes data]
* (304) (IN), TLS handshake, Certificate (11):
{ [2308 bytes data]
* (304) (IN), TLS handshake, CERT verify (15):
{ [80 bytes data]
* (304) (IN), TLS handshake, Finished (20):
{ [52 bytes data]
* (304) (OUT), TLS handshake, Finished (20):
} [52 bytes data]
* SSL connection using TLSv1.3 / AEAD-AES256-GCM-SHA384
* ALPN: server accepted http/1.1
* Server certificate:
*  subject: C=US; ST=California; L=San Francisco; O=Cloudflare, Inc.; CN=cdn-updates.orbstack.dev
*  start date: Feb 15 00:00:00 2023 GMT
*  expire date: Feb 14 23:59:59 2024 GMT
*  subjectAltName: host "cdn-updates.orbstack.dev" matched cert's "cdn-updates.orbstack.dev"
*  issuer: C=US; O=Cloudflare, Inc.; CN=Cloudflare Inc ECC CA-3
*  SSL certificate verify ok.
* using HTTP/1.1
> GET /amd64/OrbStack_v0.11.3_1648_amd64.dmg HTTP/1.1
> Host: cdn-updates.orbstack.dev
> User-Agent: curl/7.88.1
> Accept: */*
>
< HTTP/1.1 200 OK
< Date: Thu, 08 Jun 2023 03:11:20 GMT
< Content-Type: application/x-apple-diskimage
< Content-Length: 188568660
< Connection: keep-alive
< ETag: "d41ea20589a579e6539534c6496d9c9b"
< Last-Modified: Wed, 07 Jun 2023 08:48:08 GMT
< Vary: Accept-Encoding
< Cache-Control: max-age=14400
< CF-Cache-Status: HIT
< Age: 1212
< Accept-Ranges: bytes
< Report-To: {"endpoints":[{"url":"https://a.nel.cloudflare.com/report/v3?s=oS23nKTH22MjE3MDa9BsDWT%2FJjhm1Fvd76jv6tpGgL6NUxqVgIVQsYRDJn3ULFl2UAgbXOGt%2BefaI3n1%2FDYLwxBYoGtbCKo9E7HELYlm%2BKmXkKUgJstXQDV884ny3Kh09jPBJSrOMsW0ludbE4%2FBTanA0PXWQrM%3D"}],"group":"cf-nel","max_age":604800}
< NEL: {"success_fraction":0,"report_to":"cf-nel","max_age":604800}
< Server: cloudflare
< CF-RAY: 7d3df26aacf8187d-EWR
< alt-svc: h3=":443"; ma=86400
<
{ [584 bytes data]
* Connection #0 to host cdn-updates.orbstack.dev left intact

2023-06-07 23:11:26 : REQ   : orbstack : no more blocking processes, continue with update
2023-06-07 23:11:26 : REQ   : orbstack : Installing OrbStack
2023-06-07 23:11:26 : INFO  : orbstack : Mounting /var/folders/zz/zyxvpxvq6csfxvn_n0000000000000/T/tmp.nZyoVqJb/OrbStack.dmg
2023-06-07 23:11:29 : DEBUG : orbstack : Debugging enabled, dmgmount output was:
Checksumming Protective Master Boot Record (MBR : 0)…
Protective Master Boot Record (MBR :: verified   CRC32 $9074DE8A
Checksumming GPT Header (Primary GPT Header : 1)…
GPT Header (Primary GPT Header : 1): verified   CRC32 $DE20DD62
Checksumming GPT Partition Data (Primary GPT Table : 2)…
GPT Partition Data (Primary GPT Tabl: verified   CRC32 $3274CF48
Checksumming  (Apple_Free : 3)…
(Apple_Free : 3): verified   CRC32 $00000000
Checksumming disk image (Apple_APFS : 4)…
disk image (Apple_APFS : 4): verified   CRC32 $0DEF00EA
Checksumming  (Apple_Free : 5)…
(Apple_Free : 5): verified   CRC32 $00000000
Checksumming GPT Partition Data (Backup GPT Table : 6)…
GPT Partition Data (Backup GPT Table: verified   CRC32 $3274CF48
Checksumming GPT Header (Backup GPT Header : 7)…
GPT Header (Backup GPT Header : 7): verified   CRC32 $51DC02B9
verified   CRC32 $4A21B271
/dev/disk2          	GUID_partition_scheme
/dev/disk2s1        	Apple_APFS
/dev/disk4          	EF57347C-0000-11AA-AA11-0030654
/dev/disk4s1        	41504653-0000-11AA-AA11-0030654	/Volumes/Install OrbStack v0.11.3

2023-06-07 23:11:29 : INFO  : orbstack : Mounted: /Volumes/Install OrbStack v0.11.3
2023-06-07 23:11:29 : INFO  : orbstack : Verifying: /Volumes/Install OrbStack v0.11.3/OrbStack.app
2023-06-07 23:11:29 : DEBUG : orbstack : App size: 325M	/Volumes/Install OrbStack v0.11.3/OrbStack.app
2023-06-07 23:11:33 : DEBUG : orbstack : Debugging enabled, App Verification output was:
/Volumes/Install OrbStack v0.11.3/OrbStack.app: accepted
source=Notarized Developer ID
origin=Developer ID Application: Orbital Labs, LLC (U.S.) (HUAQ24HBR6)

2023-06-07 23:11:33 : INFO  : orbstack : Team ID matching: HUAQ24HBR6 (expected: HUAQ24HBR6 )
2023-06-07 23:11:33 : INFO  : orbstack : Downloaded version of OrbStack is 0.11.3 on versionKey CFBundleShortVersionString, same as installed.
2023-06-07 23:11:33 : DEBUG : orbstack : Unmounting /Volumes/Install OrbStack v0.11.3
2023-06-07 23:11:33 : DEBUG : orbstack : Debugging enabled, Unmounting output was:
"disk2" ejected.
2023-06-07 23:11:33 : DEBUG : orbstack : Deleting /var/folders/zz/zyxvpxvq6csfxvn_n0000000000000/T/tmp.nZyoVqJb
2023-06-07 23:11:33 : DEBUG : orbstack : Debugging enabled, Deleting tmpDir output was:
/var/folders/zz/zyxvpxvq6csfxvn_n0000000000000/T/tmp.nZyoVqJb/OrbStack.dmg
2023-06-07 23:11:33 : DEBUG : orbstack : /var/folders/zz/zyxvpxvq6csfxvn_n0000000000000/T/tmp.nZyoVqJb
2023-06-07 23:11:33 : INFO  : orbstack : App not closed, so no reopen.
2023-06-07 23:11:33 : REG   : orbstack : No new version to install
2023-06-07 23:11:33 : REQ   : orbstack : ################## End Installomator, exit code 0

scriptingosx commented:
Getting an error now, looks like the download page changed?

```
➜ ./assemble.sh orbstack
2023-08-22 19:12:06 : REQ   : orbstack : ################## Start Installomator v. 10.5beta, date 2023-08-22
2023-08-22 19:12:06 : INFO  : orbstack : ################## Version: 10.5beta
2023-08-22 19:12:06 : INFO  : orbstack : ################## Date: 2023-08-22
2023-08-22 19:12:06 : INFO  : orbstack : ################## orbstack
2023-08-22 19:12:06 : DEBUG : orbstack : DEBUG mode 1 enabled.
Download URL: 
2023-08-22 19:12:06 : ERROR : orbstack : need to provide 'downloadURL'
```

jnichols35 commented:
@scriptingosx I fixed the download code, though I'm not sure if it's the right thing to do. 

I used `curl -fs` originally, and it was fine. I'm not sure what changed, but I switched it to `curl -fsI` and it grabs the latest version. 

```
./Installomator.sh orbstack DEBUG=0
2023-08-30 18:09:54 : REQ   : orbstack : ################## Start Installomator v. 10.5beta, date 2023-08-30
2023-08-30 18:09:54 : INFO  : orbstack : ################## Version: 10.5beta
2023-08-30 18:09:54 : INFO  : orbstack : ################## Date: 2023-08-30
2023-08-30 18:09:54 : INFO  : orbstack : ################## orbstack
2023-08-30 18:09:54 : DEBUG : orbstack : DEBUG mode 1 enabled.
2023-08-30 18:09:54 : INFO  : orbstack : SwiftDialog is not installed, clear cmd file var
2023-08-30 18:09:55 : INFO  : orbstack : setting variable from argument DEBUG=0
2023-08-30 18:09:55 : DEBUG : orbstack : name=OrbStack
2023-08-30 18:09:55 : DEBUG : orbstack : appName=
2023-08-30 18:09:55 : DEBUG : orbstack : type=dmg
2023-08-30 18:09:55 : DEBUG : orbstack : archiveName=
2023-08-30 18:09:55 : DEBUG : orbstack : downloadURL=https://cdn-updates.orbstack.dev/amd64/OrbStack_v0.17.0_15946_amd64.dmg
2023-08-30 18:09:55 : DEBUG : orbstack : curlOptions=
2023-08-30 18:09:55 : DEBUG : orbstack : appNewVersion=
2023-08-30 18:09:55 : DEBUG : orbstack : appCustomVersion function: Not defined
2023-08-30 18:09:55 : DEBUG : orbstack : versionKey=CFBundleShortVersionString
2023-08-30 18:09:55 : DEBUG : orbstack : packageID=
2023-08-30 18:09:55 : DEBUG : orbstack : pkgName=
2023-08-30 18:09:55 : DEBUG : orbstack : choiceChangesXML=
2023-08-30 18:09:55 : DEBUG : orbstack : expectedTeamID=HUAQ24HBR6
2023-08-30 18:09:55 : DEBUG : orbstack : blockingProcesses=
2023-08-30 18:09:55 : DEBUG : orbstack : installerTool=
2023-08-30 18:09:55 : DEBUG : orbstack : CLIInstaller=
2023-08-30 18:09:55 : DEBUG : orbstack : CLIArguments=
2023-08-30 18:09:55 : DEBUG : orbstack : updateTool=
2023-08-30 18:09:55 : DEBUG : orbstack : updateToolArguments=
2023-08-30 18:09:55 : DEBUG : orbstack : updateToolRunAsCurrentUser=
2023-08-30 18:09:55 : INFO  : orbstack : BLOCKING_PROCESS_ACTION=tell_user
2023-08-30 18:09:55 : INFO  : orbstack : NOTIFY=success
2023-08-30 18:09:55 : INFO  : orbstack : LOGGING=DEBUG
2023-08-30 18:09:55 : INFO  : orbstack : LOGO=/System/Applications/App Store.app/Contents/Resources/AppIcon.icns
2023-08-30 18:09:55 : INFO  : orbstack : Label type: dmg
2023-08-30 18:09:55 : INFO  : orbstack : archiveName: OrbStack.dmg
2023-08-30 18:09:55 : INFO  : orbstack : no blocking processes defined, using OrbStack as default
2023-08-30 18:09:55 : DEBUG : orbstack : Changing directory to /var/folders/8r/47h0wdk97rbgk74g4r93q0wr0000gp/T/tmp.TKGWGZxM
2023-08-30 18:09:55 : INFO  : orbstack : App(s) found: /Applications/OrbStack.app
2023-08-30 18:09:55 : INFO  : orbstack : found app at /Applications/OrbStack.app, version 0.11.3, on versionKey CFBundleShortVersionString
2023-08-30 18:09:55 : INFO  : orbstack : appversion: 0.11.3
2023-08-30 18:09:55 : INFO  : orbstack : Latest version not specified.
2023-08-30 18:09:55 : REQ   : orbstack : Downloading https://cdn-updates.orbstack.dev/amd64/OrbStack_v0.17.0_15946_amd64.dmg to OrbStack.dmg
2023-08-30 18:09:55 : DEBUG : orbstack : No Dialog connection, just download
2023-08-30 18:10:21 : DEBUG : orbstack : File list: -rw-r--r--  1 jakenichols  staff   264M Aug 30 18:10 OrbStack.dmg
2023-08-30 18:10:21 : DEBUG : orbstack : File type: OrbStack.dmg: lzfse encoded, lzvn compressed
2023-08-30 18:10:21 : DEBUG : orbstack : curl output was:
*   Trying [2606:4700:3037::6815:3732]:443...
* Connected to cdn-updates.orbstack.dev (2606:4700:3037::6815:3732) port 443 (#0)
* ALPN: offers h2,http/1.1
* (304) (OUT), TLS handshake, Client hello (1):
} [329 bytes data]
*  CAfile: /etc/ssl/cert.pem
*  CApath: none
* (304) (IN), TLS handshake, Server hello (2):
{ [122 bytes data]
* (304) (IN), TLS handshake, Unknown (8):
{ [25 bytes data]
* (304) (IN), TLS handshake, Certificate (11):
{ [2308 bytes data]
* (304) (IN), TLS handshake, CERT verify (15):
{ [79 bytes data]
* (304) (IN), TLS handshake, Finished (20):
{ [52 bytes data]
* (304) (OUT), TLS handshake, Finished (20):
} [52 bytes data]
* SSL connection using TLSv1.3 / AEAD-AES256-GCM-SHA384
* ALPN: server accepted http/1.1
* Server certificate:
*  subject: C=US; ST=California; L=San Francisco; O=Cloudflare, Inc.; CN=cdn-updates.orbstack.dev
*  start date: Feb 15 00:00:00 2023 GMT
*  expire date: Feb 14 23:59:59 2024 GMT
*  subjectAltName: host "cdn-updates.orbstack.dev" matched cert's "cdn-updates.orbstack.dev"
*  issuer: C=US; O=Cloudflare, Inc.; CN=Cloudflare Inc ECC CA-3
*  SSL certificate verify ok.
* using HTTP/1.1
> GET /amd64/OrbStack_v0.17.0_15946_amd64.dmg HTTP/1.1
> Host: cdn-updates.orbstack.dev
> User-Agent: curl/8.1.2
> Accept: */*
>
< HTTP/1.1 200 OK
< Date: Wed, 30 Aug 2023 22:09:56 GMT
< Content-Type: application/x-apple-diskimage
< Content-Length: 276997361
< Connection: keep-alive
< ETag: "b4f5d8b8560c5c875408d0c18b6921f3-53"
< Last-Modified: Tue, 29 Aug 2023 07:26:54 GMT
< Vary: Accept-Encoding
< Cache-Control: max-age=14400
< CF-Cache-Status: HIT
< Age: 619
< Accept-Ranges: bytes
< Report-To: {"endpoints":[{"url":"https://a.nel.cloudflare.com/report/v3?s=VI4NQ7hiLMxNocakj%2BAvKIgT6XfrZFEtAWci6HPnfjamyNz633fQMQUJ9h9hUDPXOwA2mPBzE98bGJSos4xYkQhiZZHiJzG5xMIhGPxUBSgEfQzy6sJlvho5D%2F8f8kgHuEkJ%2B8KwecS6uLrXSLSsg671g0HPeS0%3D"}],"group":"cf-nel","max_age":604800}
< NEL: {"success_fraction":0,"report_to":"cf-nel","max_age":604800}
< Server: cloudflare
< CF-RAY: 7ff05c652a39424f-EWR
< alt-svc: h3=":443"; ma=86400
<
{ [586 bytes data]
* Connection #0 to host cdn-updates.orbstack.dev left intact

2023-08-30 18:10:22 : REQ   : orbstack : no more blocking processes, continue with update
2023-08-30 18:10:22 : REQ   : orbstack : Installing OrbStack
2023-08-30 18:10:22 : INFO  : orbstack : Mounting /var/folders/8r/47h0wdk97rbgk74g4r93q0wr0000gp/T/tmp.TKGWGZxM/OrbStack.dmg
2023-08-30 18:10:24 : DEBUG : orbstack : Debugging enabled, dmgmount output was:
Checksumming Protective Master Boot Record (MBR : 0)…
Protective Master Boot Record (MBR :: verified   CRC32 $1ADF953E
Checksumming GPT Header (Primary GPT Header : 1)…
GPT Header (Primary GPT Header : 1): verified   CRC32 $4BDC1E49
Checksumming GPT Partition Data (Primary GPT Table : 2)…
GPT Partition Data (Primary GPT Tabl: verified   CRC32 $D88B756C
Checksumming  (Apple_Free : 3)…
(Apple_Free : 3): verified   CRC32 $00000000
Checksumming disk image (Apple_APFS : 4)…
disk image (Apple_APFS : 4): verified   CRC32 $18237EF1
Checksumming  (Apple_Free : 5)…
(Apple_Free : 5): verified   CRC32 $00000000
Checksumming GPT Partition Data (Backup GPT Table : 6)…
GPT Partition Data (Backup GPT Table: verified   CRC32 $D88B756C
Checksumming GPT Header (Backup GPT Header : 7)…
GPT Header (Backup GPT Header : 7): verified   CRC32 $7A341BBD
verified   CRC32 $0B95755C
/dev/disk2          	GUID_partition_scheme
/dev/disk2s1        	Apple_APFS
/dev/disk3          	EF57347C-0000-11AA-AA11-0030654
/dev/disk3s1        	41504653-0000-11AA-AA11-0030654	/Volumes/Install OrbStack v0.17.0

2023-08-30 18:10:24 : INFO  : orbstack : Mounted: /Volumes/Install OrbStack v0.17.0
2023-08-30 18:10:24 : INFO  : orbstack : Verifying: /Volumes/Install OrbStack v0.17.0/OrbStack.app
2023-08-30 18:10:24 : DEBUG : orbstack : App size: 480M	/Volumes/Install OrbStack v0.17.0/OrbStack.app
2023-08-30 18:10:29 : DEBUG : orbstack : Debugging enabled, App Verification output was:
/Volumes/Install OrbStack v0.17.0/OrbStack.app: accepted
source=Notarized Developer ID
origin=Developer ID Application: Orbital Labs, LLC (U.S.) (HUAQ24HBR6)

2023-08-30 18:10:29 : INFO  : orbstack : Team ID matching: HUAQ24HBR6 (expected: HUAQ24HBR6 )
2023-08-30 18:10:29 : INFO  : orbstack : Downloaded version of OrbStack is 0.17.0 on versionKey CFBundleShortVersionString (replacing version 0.11.3).
2023-08-30 18:10:29 : INFO  : orbstack : App has LSMinimumSystemVersion: 12.3
2023-08-30 18:10:29 : WARN  : orbstack : Removing existing /Applications/OrbStack.app
2023-08-30 18:10:30 : DEBUG : orbstack : Debugging enabled, App removing output was:
/Applications/OrbStack.app/Contents/
Last Log repeated 286 times
/Applications/OrbStack.app/Contents
/Applications/OrbStack.app

2023-08-30 18:10:30 : INFO  : orbstack : Copy /Volumes/Install OrbStack v0.17.0/OrbStack.app to /Applications
2023-08-30 18:10:30 : DEBUG : orbstack : Debugging enabled, App copy output was:
Copying /Volumes/Install OrbStack v0.17.0/OrbStack.app

2023-08-30 18:10:30 : WARN  : orbstack : Changing owner to jakenichols
2023-08-30 18:10:30 : INFO  : orbstack : Finishing...
2023-08-30 18:10:33 : INFO  : orbstack : App(s) found: /Applications/OrbStack.app
2023-08-30 18:10:34 : INFO  : orbstack : found app at /Applications/OrbStack.app, version 0.17.0, on versionKey CFBundleShortVersionString
2023-08-30 18:10:34 : REQ   : orbstack : Installed OrbStack, version 0.17.0
2023-08-30 18:10:34 : INFO  : orbstack : notifying
displaynotification:7: no such file or directory: /usr/local/bin/dialog
2023-08-30 18:10:34 : DEBUG : orbstack : Unmounting /Volumes/Install OrbStack v0.17.0
2023-08-30 18:10:34 : DEBUG : orbstack : Debugging enabled, Unmounting output was:
"disk2" ejected.
2023-08-30 18:10:34 : DEBUG : orbstack : Deleting /var/folders/8r/47h0wdk97rbgk74g4r93q0wr0000gp/T/tmp.TKGWGZxM
2023-08-30 18:10:34 : DEBUG : orbstack : Debugging enabled, Deleting tmpDir output was:
/var/folders/8r/47h0wdk97rbgk74g4r93q0wr0000gp/T/tmp.TKGWGZxM/OrbStack.dmg
2023-08-30 18:10:34 : DEBUG : orbstack : /var/folders/8r/47h0wdk97rbgk74g4r93q0wr0000gp/T/tmp.TKGWGZxM
2023-08-30 18:10:34 : INFO  : orbstack : App not closed, so no reopen.
2023-08-30 18:10:34 : REQ   : orbstack : All done!
2023-08-30 18:10:34 : REQ   : orbstack : ################## End Installomator, exit code 0
```

scriptingosx commented:
This can be simplified down to:

```
orbstack)
    name="OrbStack"
    type="dmg"
    if [[ "$(arch)" == "arm64" ]]; then
        downloadURL="https://orbstack.dev/download/stable/latest/arm64"
    else
        downloadURL="https://orbstack.dev/download/stable/latest/amd64"
    fi
    expectedTeamID="HUAQ24HBR6"
    ;;
```

Also, your URL points to the _beta_ version, not the _stable_ version. We usually don't link to betas with Installomator. And if that is a requirement, then create a second label with 'beta'.

Also there is no `appNewVersion`
